### ISSUE TITLE
How to flip image horizontally/vertically?

### ISSUE THREAD

**Issue by qweluke** [liip/LiipImagineBundle#918]
| Q | A
| --- | ---
| Bug Report? | no
| Feature Request? | no
| BC Break Report? | no
| RFC? | no
| Imagine Bundle Version | ^1.6

quick question: how to flip image vertically and/or horizontally?

**Comment by robfrawley** [comment]
You must implement a [custom filter](http://symfony.com/doc/current/bundles/LiipImagineBundle/filters.html#custom-filters) as none of the filters provided by default implement a flip method. For example, here is the basic filter implementation you'd need:

```php
<?php

namespace AppBundle\Imagine\Filter\Loader;

use Imagine\Image\ImageInterface;
use Liip\ImagineBundle\Exception\InvalidArgumentException;
use Liip\ImagineBundle\Imagine\Filter\Loader\LoaderInterface;
use Symfony\Component\OptionsResolver\OptionsResolver;
use Symfony\Component\OptionsResolver\Exception\ExceptionInterface;

class FlipFilter implements LoaderInterface
{
    /**
     * @param ImageInterface $image
     * @param array          $options
     *
     * @return ImageInterface
     */
    public function load(ImageInterface $image, array $options = array())
    {
        try {
            $options = (new OptionsResolver())
                ->setDefault('axis', 'x')
                ->setAllowedValues('axis', ['x', 'y'])
                ->resolve($options);
        } catch (ExceptionInterface $e) {
            throw new InvalidArgumentException('The "axis" option must be set to "x" or "y"', null, $e);
        }

        if ($options['axis'] === 'x') {
            $image->flipHorizontally();
        } else {
            $image->flipVertically();
        }

        return $image;
    }
}
```

Then, register the filter in your `services.yml` file, ensuring it is tagged with the "name" `liip_imagine.filter.loader` and the "loader" set to whatever name you'd like to reference the filter from your filter set configuration.

```yml
# app/config/services.yml

services:
    app.filter.flip_filter:
        class: AppBundle\Imagine\Filter\Loader\FlipFilter
        tags:
            - { name: "liip_imagine.filter.loader", loader: flip_filter }
```

Lastly, again, use the value of the "loader" tag to reference the filter in your configuration:

```yml
# app/config/config.yml

liip_imagine:
    filter_sets:
        my_cool_filter_set:
            filters:
                flip_filter: { axis: x }
```

**Comment by qweluke** [comment]
@robfrawley thank you!

**Comment by robfrawley** [comment]
FYI: I just submitted a pull request (#920) that adds the `flip` filter (identical to my above example) within the core bundle; it should be included in the next minor release, likely `1.7.5`.

**Comment by robfrawley** [comment]
The above-detailed functionality has been merged into the core in PR #920, you can use it immediately by requiring `dev-master` in your composer file, or you can wait for the next release `1.7.5`. Thanks for finding this notable shortcoming in the default provided filter implementations @qweluke!
